### PR TITLE
Add Event UI

### PR DESCRIPTION
**Pull request by leslieyip02**
## Description

- [x] Create `EventCard`
- [x] Update logic to render events in the app's UI
- [x] Modified `ListCommand` such that it clears the `ModelManager`'s `filteredEvents`
- [x] Modified `ListEventCommand` such that it clears the `ModelManager`'s `filteredContacts`

### Demo
Screenshot 2024-10-24 130723

### UML
image

### CODE REVIEW

**Review by casperngeen — CHANGES_REQUESTED** [review]
Only the grammatical error in the rendering needs to be changed, the rest looks good

File: src/main/java/seedu/ddd/logic/commands/ListEventCommand.java [hunk 1]
```diff
@@ -28,10 +32,12 @@ public ListEventCommand(Predicate<Event> predicate) {
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
+        model.updateFilteredContactList(CLEAR_CONTACTS);
         model.updateFilteredEventList(predicate);
         return new CommandResult(String.format(MESSAGE_EVENTS_LISTED_OVERVIEW,
```
[the remark this answers is not in the document]
Comment (leslieyip02) [reply src/main/java/seedu/ddd/logic/commands/ListEventCommand.java:37]: This bug also applies for listing contacts, clients and vendors.

**Review by Teddayz — COMMENTED** [review]
LGTM, doesn't look like theres anything wrong. Will test it out once we merge.

File: src/main/java/seedu/ddd/logic/commands/ListEventCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Teddayz) [reply src/main/java/seedu/ddd/logic/commands/ListEventCommand.java:37]: Sorry, whats the bug?

[the remark this answers is not in the document]
Comment (leslieyip02) [reply src/main/java/seedu/ddd/logic/commands/ListEventCommand.java:37]: Changed messages to support both singular/plural.

```java
    ...
    public static final String MESSAGE_CONTACTS_LISTED_OVERVIEW = "%1$d contact(s) listed!";
    public static final String MESSAGE_EVENTS_LISTED_OVERVIEW = "%1$d event(s) listed!";
    public static final String MESSAGE_CLIENTS_LISTED_OVERVIEW = "%1$d client(s) listed!";
    public static final String MESSAGE_VENDORS_LISTED_OVERVIEW = "%1$d vendor(s) listed!";
    ...
```

**Pull request merged**